src/cert/mod.rs

```diff
@@ -109,6 +114,168 @@ pub async fn upload_new_cert_ref(
+        Err(e) => {
+            log::error!("Error getting cage signing cert refs — {:?}", e);
+            return Err(CertError::ApiError(e));
+        }
+    };
+
+    let locked_certs = match cage_api.get_cage_locked_signing_certs(cage_uuid).await {
+        Ok(certs) => certs,
+        Err(e) => {
+            log::error!("Error getting cage signing cert — {:?}", e);
+            return Err(CertError::ApiError(e));
+        }
+    };
+
+    let locked_cert_uuids = locked_certs
+        .iter()
+        .map(|cert| cert.uuid())
+        .collect::<Vec<&str>>();
+
+    let available_stripped: Vec<&CageSigningCert> = available_certs
+        .iter()
+        .filter(|cert| !locked_cert_uuids.contains(&cert.uuid()))
+        .collect();
+
+    let available_formatted = available_stripped
+        .iter()
+        .map(|cert| CertWithFormattedString::new(cert, false))
+        .collect::<Vec<CertWithFormattedString>>();
+    let locked_formatted = locked_certs
+        .iter()
+        .map(|cert| CertWithFormattedString::new(cert, true))
+        .collect::<Vec<CertWithFormattedString>>();
+
+    let mut all_formatted: Vec<CertWithFormattedString> = vec![];
+
+    all_formatted.extend(available_formatted);
+    all_formatted.extend(locked_formatted);
+
+    Ok(all_formatted)
+}
+
+fn sort_certs_by_expiry(
+    mut certs: Vec<CertWithFormattedString>,
+) -> Result<Vec<CertWithFormattedString>, CertError> {
+    certs.sort_by_key(|cert| cert.cert.not_after().unwrap_or("".to_string()));
+    Ok(certs)
+}
+
+pub async fn lock_cage_to_certs(api_key: &str, cage_uuid: &str) -> Result<(), CertError> {
+    let cage_api = api::cage::CagesClient::new(AuthMode::ApiKey(api_key.to_string()));
+
+    let certs_for_select = get_certs_for_selection(cage_api.clone(), cage_uuid).await?;
+
+    let sorted_certs_for_select = sort_certs_by_expiry(certs_for_select)?;
+
+    let chosen: Vec<usize> = MultiSelect::new()
+        .with_prompt("Select Certs To Lock Cage To. Press Space To Select, Enter To Confirm.\n Cert Name | PCR8 (Hash of cert) | Cert Expiry ")
+        .report(false)
+        .max_length(6)
+        .items_checked(
+            sorted_certs_for_select
+                .iter()
+                .map(|cert| (cert.formatted.as_str(), cert.locked))
+                .collect::<Vec<(&str, bool)>>()
+                .as_slice(),
+        )
+        .interact()?;
+
+    let chosen_cert_uuids = chosen
+        .iter()
+        .map(|index| {
+            sorted_certs_for_select
+                .get(*index)
+                .unwrap()
+                .cert
+                .uuid()
+                .to_string()
+        })
+        .collect::<Vec<String>>();
+
+    let payload = UpdateLockedCageSigningCertRequest::new(chosen_cert_uuids.clone());
+
+    let amount_chosen = chosen_cert_uuids.len();
+    let msg = match amount_chosen {
+        0 => format!(
+            "No certs selected. Cage {} will not be locked to any certs.",
+            cage_uuid
```

Comment:
might to use cage name here for readability. Also, should we confirm if someone is removing existing locks and setting to zero certs? Feels like a dangerous thing to allow without a second confirmation

Reply:
Good shout, updated to alway confirm after printing this message